engine/src/main/java/org/terasology/world/block/family/MultiConnectBlockFamily.java

```diff
@@ -175,9 +179,53 @@ private Rotation getRotationToAchieve(byte source, byte target) {
         return null;
     }
 
+
+    @Override
+    public Block getArchetypeBlock() {
+        return archetypeBlock;
+    }
+
+    @Override
+    public Block getBlockForPlacement(Vector3i location, Side attachmentSide, Side direction) {
+        byte connections = 0;
+        for (Side connectSide : SideBitFlag.getSides(getConnectionSides())) {
+            if (this.connectionCondition(location, connectSide)) {
+                connections += SideBitFlag.getSide(connectSide);
+            }
+        }
+        return blocks.get(connections);
+    }
+
+    public Block getBlockForNeighborUpdate(Vector3i location, Block oldBlock) {
+        byte connections = 0;
+        for (Side connectSide : SideBitFlag.getSides(getConnectionSides())) {
+            if (this.connectionCondition(location, connectSide)) {
+                connections += SideBitFlag.getSide(connectSide);
+            }
+        }
+        return blocks.get(connections);
+    }
+
+    protected boolean connectionCondition(Vector3i blockLocation, Side connectSide) {
+        return false;
+    }
+
+    @Override
+    public Block getBlockFor(BlockUri blockUri) {
+        if (getURI().equals(blockUri.getFamilyUri())) {
+            try {
+                byte connections = Byte.parseByte(blockUri.getIdentifier().toString().toLowerCase(Locale.ENGLISH));
+                return blocks.get(connections);
+            } catch (IllegalArgumentException e) {
+                return null;
```

Comment:
please log (throwing seems better but definitely don't silently eat the exception)